packages/api/src/index.js

```diff
@@ -23,12 +23,27 @@ const router = Router()
 router.options('*', corsOptions)
 router.all('*', envAll)
 
+/**
+ * Takes any number of functions and invokes them all one after the other
+ *
+ * @param  {...function(...any):any} fns
+ * @returns
+ */
+const compose = (...fns) => x => fns.reduceRight((y, f) => f(y), x)
+
+/**
+ * It defines a list of "middlewares" that need to be applied for a given authentication mode.
+ * Each value takes an endpoint handler and returns a "composed" version.
+ *
+ * @type {Object.<string, function(...any):any>}
+ */
 const auth = {
-  '🤲': handler => withCorsHeaders(handler),
-  '🔒': handler => withCorsHeaders(withApiOrMagicToken(handler)),
-  '👮': handler => withCorsHeaders(withMagicToken(handler)),
-  '📌': handler => withPinningAuthorized(handler),
-  '🚫': handler => withAccountNotRestricted(handler)
+  '🤲': withCorsHeaders,
+  '🔒': compose(withCorsHeaders, withApiOrMagicToken),
+  '👮': compose(withCorsHeaders, withMagicToken),
+  '🚫': compose(withCorsHeaders, withApiOrMagicToken, withAccountNotRestricted),
+  '👀📌': compose(withCorsHeaders, withApiOrMagicToken, withPinningAuthorized),
+  '📝📌': compose(withCorsHeaders, withApiOrMagicToken, withAccountNotRestricted, withPinningAuthorized)
```

Comment:
Let's revisit these icons! I don't think they are making things clearer

- routes that _allow_ a *blocked* user to do a thing are definitly a _special case_ and so should visually stand out.

```suggestion
  '🌍': withCorsHeaders,                                        // world readable!                      
  '🔑⚠️': compose(withCorsHeaders, withApiOrMagicToken),        // any key will do & blocked users allowed!
  '👤': compose(withCorsHeaders, withMagicToken),               // must be a logged in user
  '🔑': compose(withCorsHeaders, withApiOrMagicToken, withAccountNotRestricted), // any key will do.

  '📌⚠️': compose(withCorsHeaders, withApiOrMagicToken, withPinningAuthorized), // needs PSA & blocked users allowed
  '📌': compose(withCorsHeaders, withApiOrMagicToken, withAccountNotRestricted, withPinningAuthorized)  // needs PSA
```

Reply:
Thanks, this is better. 
I still have a feeling this might be not obvious to someone new, but definitively a great improv from what we had.

Thanks Olly!